tests/ui/invalid_deprecated_pyfunction.rs

```diff
@@ -0,0 +1,27 @@
+use pyo3::prelude::*;
+
+#[pyfunction]
+#[pyo3(deprecated)]
+fn no_parenthesis_deprecated() {}
+
+#[pyfunction]
+#[pyo3(deprecated())]
+fn no_message_deprecated() {}
+
+#[pyfunction]
+#[pyo3(deprecated(category = pyo3::exceptions::PyDeprecationWarning))]
+fn no_message_deprecated_with_category() {}
+
+#[pyfunction]
+#[pyo3(deprecated(category = pyo3::exceptions::PyDeprecationWarning, message = ,))]
+fn empty_message_deprecated_with_category() {}
+
+#[pyfunction]
+#[pyo3(deprecated(message = "deprecated function", category = ,))]
+fn empty_category_deprecated_with_message() {}
+
+#[pyfunction]
+#[pyo3(deprecated(message = "deprecated function", random_key))]
+fn random_key_deprecated() {}
+
+fn main() {}
```

Comment:
If you're developing on Windows, you need to make these files end with a blank line ;)

Reply:
Oh that's interesting and good to know :)